File: E-Commerce-Project-Backend/src/main/java/org/sid/ecommerce/Entities/Product.java
package org.sid.ecommerce.Entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;

import lombok.*;

@Entity @NoArgsConstructor @AllArgsConstructor
@Data
public class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;

    private int price;
    private int quantity;

    private String imageName;
    private String imageType;
    @Lob
    private byte[] imageDate;


    // Utiliser @JsonBackReference pour ignorer cette référence dans la sérialisation
    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "category_id")
    private org.sid.ecommerce.Entities.Category category;
}

File: E-Commerce-Project-Xamarin/ShopProject/ShopProject/ViewModel/Product.cs
   public class Product
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Price { get; set; }
        public int Quantity { get; set; }
        public string ImageName { get; set; }
        public long CategoryId { get; set; } 

        // Property to get the image source for Xamarin
        public string ImageSource => $"resource://{ImageName}";
    }